apps/home/test/routes.coffee

```diff
@@ -0,0 +1,116 @@
+          Backbone.sync
+            .onCall 0
+            .yieldsTo 'success', [fabricate 'site_hero_unit']
+            .onCall 1
+            .yieldsTo 'success', [fabricate 'featured_link']
+            .onCall 2
+            .yieldsTo 'success', [fabricate 'featured_link']
+
+          routes.index @req, @res
+            .then =>
+              @res.render.args[0][0].should.equal 'index'
+              @res.render.args[0][1].heroUnits.at(0).get 'description'
+                .should.equal 'Sign up to get updates on your favorite artists'
+              @res.render.args[0][1].heroUnits.at(1).get 'description'
+                .should.equal 'My hero'
+              @res.cookie.args[0][0].should.equal 'hide-welcome-hero'
+              @res.cookie.args[0][1].should.equal '1'
+
+      describe 'after first time', ->
+        it 'renders the home page with hero units + welcome hero unit at the end', ->
+          @req.cookies = 'hide-welcome-hero': '1'
+
+          Backbone.sync
+            .onCall 0
+            .yieldsTo 'success', [fabricate 'site_hero_unit']
+            .onCall 1
+            .yieldsTo 'success', [fabricate 'featured_link']
+            .onCall 2
+            .yieldsTo 'success', [fabricate 'featured_link']
+
+          routes.index @req, @res
+            .then =>
+              @res.render.args[0][0].should.equal 'index'
+              @res.render.args[0][1].heroUnits.last().get 'description'
+                .should.equal 'Sign up to get updates on your favorite artists'
+
+    describe 'logged in', ->
+      it 'renders the homepage without the welcome hero unit', ->
+        Backbone.sync
+          .onCall 0
+          .yieldsTo 'success', [fabricate 'site_hero_unit']
+          .onCall 1
+          .yieldsTo 'success', [fabricate 'featured_link']
+
+        routes.index extend({ user: 'existy' }, @req), @res
+          .then =>
+            @res.render.args[0][0].should.equal 'index'
+            @res.render.args[0][1].heroUnits.first().get 'description'
+              .should.equal 'My hero'
+
+  describe '#redirectToSignup', ->
+    it 'redirects to signup', ->
+      routes.redirectToSignup @req, @res
+      @res.redirect.args[0][0].should.equal '/sign_up'
+
+  describe '#redirectLoggedInHome', ->
+    it 'redirects logged in users home', ->
+      @req.user = {}
+      routes.redirectLoggedInHome @req, @res
+      @res.redirect.args[0][0].should.equal '/'
+
+    it 'redirects logged in users to redirect location', ->
+      @req.query['redirect_uri'] = '/awesome-fair'
+      @req.user = {}
+      routes.redirectLoggedInHome @req, @res
+      @res.redirect.args[0][0].should.equal '/awesome-fair'
+
+    it 'redirects logged in users to redirect location', ->
+      @req.body['redirect-to'] = '/awesome-fair'
+      @req.user = {}
+      routes.redirectLoggedInHome @req, @res
+      @res.redirect.args[0][0].should.equal '/awesome-fair'
```

Comment:
Might want to update the description of these two tests slightly to indicate where the difference lies, took me a few sec to figure out.